common/exchange_input.go

```diff
@@ -221,3 +241,60 @@ func (sf *ServiceFile) SupportVersionRange() {
 		}
 	}
 }
+
+// Validate a service definition.
```

Comment:
I dont see code in hzn dev CLI to do this same kind of verification. It already has verification logic, so this logic should be moved there and reused by this code, or vice versa.

Reply:
Will change it when I start working on the 'hzn  dev'. There are more verification code to be added. In this one, the top level service is not in the exchange. I will add one to verify a service when all the services are in the exchange. etc. So this may not be the final place.

Reply:
I will open a new issue for pattern verification and 'hzn dev'.